src/CompositeChat.cpp

```diff
@@ -96,50 +96,52 @@ void cCompositeChat::AddShowAchievementPart(const AString & a_PlayerName, const
 
 
 
-
+/**
+* Parse the input message to add colors or link then add it to the object.
+*
+* It detect every & of the message and the next character for it to colorize.
+* It detect : in the text to detect link structures.
+*
+* @param a_ParseText The input text to parse
+*/
 void cCompositeChat::ParseText(const AString & a_ParseText)
 {
 	size_t len = a_ParseText.length();
-	size_t first = 0;  // First character of the currently parsed block
+	size_t cursor = 0;
 	AString CurrentStyle;
 	AString CurrentText;
+
 	for (size_t i = 0; i < len; i++)
 	{
 		switch (a_ParseText[i])
 		{
-			case '@':
+			case '&':  //< Color code
 			{
-				// Color code
-				i++;
-				if (i >= len)
+				if ((i != 0) && (a_ParseText[i-1] == '\\'))
```

Comment:
Does the vanilla implementation also use `\` as the escape character?

Reply:
I'm pretty sure it doesn't, but in fact it causes trouble when trying to write for instance choco&almond which would take the code &a and write chocolmond with lmond in green... I'm pretty sure there is no way to bypass this in vanilla and I figured it would kill to have this I think ^^ but ofc I can remove it if needed

Reply:
No this is a good idea. Maybe we should add this to the documentation, just in case someone is confused or wants to do this